src/supermarket/config/environments/production.rb

```diff
@@ -113,5 +113,7 @@
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
 
-  config.hosts << ENV["ALLOWED_HOST"] if ENV["ALLOWED_HOST"].present?
+  # The value of ENV["DISABLE_HOST_HEADER_ATTACK"] will be parsed as string.
+  # Hence we need to convert string to boolean.
+  config.hosts << ENV["ALLOWED_HOST"] if ActiveModel::Type::Boolean.new.cast(ENV["DISABLE_HOST_HEADER_ATTACK"]) && ENV["ALLOWED_HOST"].present?
```

Comment:
Probably breaking this line long line of conditional code - `if ActiveModel::Type::Boolean.new.cast(ENV["DISABLE_HOST_HEADER_ATTACK"]) && ENV["ALLOWED_HOST"].present?` into multiple lines will make this more readable what say ?

Reply:
Applied the suggestion